src/pages/index.js

```diff
@@ -1,129 +1,218 @@
+  api
+    .getProfileInfo()
+    .then((results) => {
+      user.setUserInfo({ name: results.name, about: results.about });
+      user.setAvatar(results.avatar);
+    })
+    .catch((error) => {
+      console.error("Caught an error: ", error.message);
+    });
+
   //functions
+  const deletePopup = new confirmationPopup("#confirmation-modal");
+  deletePopup.setEventListeners();
+
+  function addCard(name, description, template, id) {
+    const items = [{name: name, description:description}];
+    const cardsSection = new Section(
+      {
+        items,
+        renderer: (cardData) => {
+          addCard(cardData.name, cardData.link, Constants.cardTemplate, cardData._id);
+        },
+      },
+      Constants.galleryList
+    );
+    const modelCard = new Card(
+      name,
+      description,
+      template,
+      imageClickHandler,
+      (card) => {
+
+        deletePopup.open();
+        deletePopup.setDefaultSaveButton();
+        deletePopup.setSubmitAction(() => {
+          api
+            .deleteCard(card._id)
+            .then(() => {
+              card.deleteCard();
+            })
+            .then(() =>{
+              deletePopup.updateSaveButton()
+            })
+            .then(() =>{
+              deletePopup.close()
+            })
+            .catch((error) => {
+              console.error("Caught an error: ", error.message);
+            });
+        });
+      },
+      toggleLikeButton,
+      initialCardsInfo,
+      id
+    );
+
+    cardsSection.addItem(modelCard.generateCard())
+  }
 
   function createCard(name, description, template) {
-    const modelCard = new Card(name, description, template, imageClickHandler);
-    cardsSection.addItem(modelCard.addCard());
+    addPopup._submitBtn.textContent = "Saving...";
+    api
+      .addCardInfo(name, description, (name, link, id) => {
+        addCard(name, link, template, id);
+      })
+      .catch((error) => {
+        console.error("Caught an error: ", error.message);
+      })
+      .finally(() =>{
+        addPopup._submitBtn.textContent = "Save"
+      })
+      ;
   }
 
-  function handleProfileEditSubmit({ name, description }) {
-    userInfo.setUserInfo({ name: name, about: description });
-    editPopup.close();
+  function handleProfileEditSubmit(inputElements) {
+    editPopup._submitBtn.textContent = "Saving...";
+    api
+      .setUserInfo(inputElements.name, inputElements.description)
+      .then((result) => {
+        user.setUserInfo({ name: result.name, about: result.about });
+      })
+      .then(() =>{
+        editPopup.close();
+      })
+      .catch((error) => {
+        console.error("Caught an error: ", error.message);
+      })
+      .finally(() =>{
+        editPopup._submitBtn.textContent = "Save"
+      })
+      ;
   }
 
   function handleProfileAddSubmit({ name, description }) {
-    createCard(name, description, cardTemplate);
+    createCard(name, description, Constants.cardTemplate);
     addPopup.close();
```

Comment:
`  addPopup.close();`

You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well

**Please, correct it everywhere**